### DISCUSSION TITLE
Megatron Finetuning hangs with more than 1 A100 GPU

### DISCUSSION THREAD

**Discussion by dkajtoch** [NVIDIA-NeMo/NeMo#8655] [Q&A]
I am trying to run an example that looks similar to this one: https://github.com/NVIDIA/NeMo/blob/48b8204d57e59c8790aaa6eaa20384b046b1a574/examples/nlp/language_modeling/tuning/megatron_gpt_finetuning.py

I am using the Docker container `nvcr.io/nvidia/nemo:24.01.framework` with `torchrun` command. My initial model is a converted Mistral 7B to Nemo format. Execution looks like this:

```shell
torchrun --standalone --nproc-per-node 2 .../megatron_gpt_finetuning.py model.tensor_model_parallel_size=2 model.pipeline_model_parallel_size=1
```
When I use L4 GPU everything is ok (but end up with Cuda OOM) also H100 works. On the other hand, when I switch to A100 80GB initialization hangs before checkpoint loading. Below is the screenshot for L4. For A100 it hangs before (I never see a blue message) "Loading distributed checkpoint ...". Any ideas how to fix it? 
![Zrzut ekranu 2024-03-14 o 10 09 14](https://github.com/NVIDIA/NeMo/assets/32985207/2c2c0c75-b4fc-4b8f-b0fb-f7d4ad848d65)

**Comment by akoumpa** [comment]
Hi, if you try a newer container (e.g. >= 24.05) there shouldn't be an issue. If there is please re-open this ticket & apologies for the late response.
